edk2toollib/windows/capsule/inf_generator2.py

```diff
@@ -0,0 +1,197 @@
+##
+# Module to generate inf files for capsule update. Supports targeting
+# multiple ESRT nodes with the same INF.
+#
+# Copyright (c) Microsoft Corporation
+#
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+import textwrap
+
+
+class InfHeader(object):
+    def __init__(self, name, versionStr, date, arch, provider, mfgname, infstrings):
+        self.name = name
```

Comment:
curious...you lost a lot of data validation that is inf_generator.py.   Example: https://github.com/tianocore/edk2-pytool-library/blob/master/edk2toollib/windows/capsule/inf_generator.py#L110

Given that we don't know the calling stack/tool using this, this tool and its data validation is our only line of defense for making valid Microsoft Windows compliant INF files.

Reply:
Agreed with this point if it can be easily accomplished. If there are a number of "shared parameters"... maybe they could be new classes that enforce their own contents? Like a "hex version" class?

Reply:
I added data validation for much of the easy stuff; for some stuff (like description strings) I took the "if you can str() it, then it's fine"  approach. I didn't go the route of adding custom helper data classes as that seemed overkill.